Bug fix for selecting KMS key via alias ARN

BlueDragon23 opened the pull request:
When using Localstack with moto as the KMS provider, I encountered issues with selecting a key via the alias ARN. I believe this change fixes the issue by ensuring the `alias/` prefix is included when the name is extracted from the ARN. I've added tests to the KMS model to verify that, and that encryption works via all the other supported mechanisms

bblommers commented on tests/test_kms/test_model.py:
```diff
@@ -0,0 +1,52 @@
+    ciphertext, arn = backend.encrypt(key.id, PLAINTEXT, {})
+
+    assert ciphertext is not None
+    assert arn is not None
```
These assertions should be more specific. What is the format of the ARN that we're expecting?

(Same goes for all other tests)

bblommers reviewed (changes requested):
Hi @BlueDragon23, could you add another test using the boto3 SDK, to show what scenario is fixed with this change?

There are quite a few tests already in `tests/test_kms/test_kms_boto3`, so feel free to reuse or extend one of them.

BlueDragon23 commented:
Thanks for the feedback! I've updated the model tests to be more specific. It turns out there _was_ an existing test in `test_kms_boto3` that covered this use case, but it was always using the `key_id` instead of the `target_id`. I've changed that now, so it verifies the case of generating a data key using the alias ARN

bblommers reviewed (approved):
LGTM - thanks for the PR @BlueDragon23!
